From: Frank van der Linden <fllinden@amazon.com>
To: <linux-arm-kernel@lists.infradead.org>,
	<linux-kernel@vger.kernel.org>, <catalin.marinas@arm.com>,
	<will@kernel.org>
Cc: Frank van der Linden <fllinden@amazon.com>
Subject: [PATCH] arm64: vdso32: make vdso32 install conditional
Date: Thu, 27 Aug 2020 23:40:12 +0000
Message-ID: <20200827234012.19757-1-fllinden@amazon.com> (raw)

vdso32 should only be installed if CONFIG_COMPAT_VDSO is enabled,
since it's not even supposed to be compiled otherwise, and arm64
builds without a 32bit crosscompiler will fail.

Signed-off-by: Frank van der Linden <fllinden@amazon.com>
Fixes: 8d75785a8142 ("ARM64: vdso32: Install vdso32 from vdso_install")
Cc: stable@vger.kernel.org [5.4+]
---
 arch/arm64/Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index b45f0124cc16..100abbb19e0b 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -165,7 +165,8 @@ zinstall install:
 PHONY += vdso_install
 vdso_install:
 	$(Q)$(MAKE) $(build)=arch/arm64/kernel/vdso $@
-	$(Q)$(MAKE) $(build)=arch/arm64/kernel/vdso32 $@
+	$(if $(CONFIG_COMPAT_VDSO),$(Q)$(MAKE) \
+		$(build)=arch/arm64/kernel/vdso32 $@)
 
 # We use MRPROPER_FILES and CLEAN_FILES now
 archclean:
